src/raster/r.suitability.regions/r.suitability.regions.py

```diff
@@ -0,0 +1,410 @@
+#% description: Neighborhood operation (focal statistic)
+#% required: no
+#% type: string
+#% answer: median
+#% options:maximum,median,quart1,quart3
+#% guisection: Focal stats
+#%end
+
+#%option
+#% key: maxgap
+#% description: Maximum gap size (in hectares) to remove
+#% required: no
+#% type: string
+#% answer: 0
+#% key_desc: float
+#% guisection: Remove gaps
+#%end
+
+#%flag
+#% key: z
+#% label: Suitability per region
+#% description: Compute map with average suitability per region
+#% guisection: Reporting stats
+#%end
+#%flag
+#% key: a
+#% label: Calculates area of clumped areas (hectares)
+#% description: Calculates area of clumped areas
+#% guisection: Reporting stats
+#%end
+
+#%flag
+#% key: k
+#% label: Keep map with all suitable cells
+#% description: Keep map that show suitable areas (irrespective of clump size)
+#% guisection: Optional
+#%end
+
+#%flag
+#% key: f
+#% label: Keep suitability based on focal statistics
+#% description: Keep suitabiliyt based on focal statistics
+#% guisection: Optional
+#%end
+
+import sys
+import os
+import atexit
+import uuid
+import grass.script as gs
+
+RECLASSRULES1 = """\
+1:filled gaps
+2:suitable areas
+"""
+COLORRULES1 = """
+1 230:230:230
+""".strip()
+
+COLORRULES2 = """
+1 241:241:114
+2 139:205:85
+""".strip()
+
+
+CLEAN_LAY = []
+
+
+def cleanup():
+    """Remove temporary maps specified in the global list"""
+    cleanrast = list(reversed(CLEAN_LAY))
+    for rast in cleanrast:
+        ffile = gs.find_file(name=rast, element="cell", mapset=gs.gisenv()["MAPSET"])
+        if ffile["file"]:
+            gs.run_command("g.remove", flags="f", type="raster", name=rast, quiet=True)
+
+
+def tmpname(prefix):
+    """Generate a tmp name which contains prefix
+    Store the name in the global list.
+    Use only for raster maps.
+    """
+    tmpf = prefix + str(uuid.uuid4())
+    tmpf = tmpf.replace("-", "_")
+    CLEAN_LAY.append(tmpf)
+    return tmpf
+
+
+def tmpmask(input, absmin):
+    """Create tmp mask"""
+    rules = "*:{}:1".format(absmin)
+    tmprecode = tmpname("tmprecode")
+    gs.write_command(
+        "r.recode", input=input, output=tmprecode, rule="-", stdin=rules, quiet=True
+    )
+    return tmprecode
+
+
+def main(options, flags):
+
+    suitrast = options["suitrast"]
+    absminsuit = options["absminsuit"]
+    radius = int(options["radius"])
+    if (radius % 2) == 0:
+        gs.error("Radius should be an odd positive number")
+    focalstat = options["focalstat"]
+    minsuit = float(options["minsuit"])
+    minfrag = float(options["minfrag"])
+    maxgap = float(options["maxgap"])
+    suitreg = options["suitreg"]
+
+    # Flags
+    flag_c = flags["c"]
+    if flag_c:
+        c = "c"
+    else:
+        c = ""
+    flag_d = flags["d"]
+    if flag_d:
+        d = "d"
+    else:
+        d = ""
+    flag_z = flags["z"]
+    flag_k = flags["k"]
+    flag_f = flags["f"]
+    flag_a = flags["a"]
+
+    # Temporary layer names
+    tmp01 = tmpname("tmp01")
```

Comment:
Don't just number your variables. Name them.

Reply:
Another complimentary way to improve this is to create the variables only right before they are needed. This gives more context for what they are and lower likelihood of using one sooner than intended.

Reply:
Was just doing that  :-). Definitely the better way to go in most cases.